src/recipes/ios-promo/index.js

```diff
@@ -0,0 +1,248 @@
+  let lastRun = extras.lastRun || eData.lastRun || 0;
+  let locale = (extras.locale || userstate.locale || "unknown").toLowerCase();
+  let restdays = config.restdays; // Only run once
+  let locales = (config.locales || []).map((x)=>x.toLowerCase());
+
+  // All versions
+
+  // Bad locale
+  console.log({
+    now: now,
+    lastRun: lastRun,
+    locale: locale,
+    restdays: restdays,
+    locales: locales,
+  })
+  if (!hasAny(locales, [locale, "*"])) {
+    events.message(NAME, "bad-locale", {locale: locale, locales: locales});
+    return false;
+  }
+  //// Already ran this
+  //if (lastRun !== 0) { // This recipe is only showed once
+  //  events.message(NAME, "already-run", {lastRun: lastRun});
+  //  return false;
+  //}
+
+  // Another survey was run recently
+  if (!waitedEnough(restdays, lastRun, now)) {
+    events.message(NAME, "too-soon", {restdays: restdays, lastRun: lastRun, now: now});
+    return false;
+  }
+  // Sample
+  let myRng = extras.randomNumber !== undefined ? extras.randomNumber : Math.random();
+
+  if (myRng <= config.sample) {
+    return true;
+  } else {
+    events.message(NAME, "bad-random-number", {randomNumber: myRng});
+    return false;
+  }
+};
+
+// run / do
+let run = function (state, extras) {
+  // TODO: We may target different messages by OS / Channel
+
+  extras = extras || {};
+  let now = extras.when || Date.now();
+  let delay = extras.delay || DELAY;
+
+  eData.data.lastRun = now;
+  eData.store();
+
+
+  let flowid = extras.flow_id || uuid();
+  let local = {
+    flow_id: flowid,
+  };
+
+  let storeFlow = function (flow_id, flow) {
+    eData.data.flows[flow_id] = flow.data;
+    eData.store();
+  }.bind(null, local.flow_id);
+
+  /* Using Telemetry instead of phonehome
+  let maybePhonehome = function (flow) {
+    if (!extras.simulate) {
+      let myRng = ;
+      if (Math.random() < PHONEHOMEPCT) {
+        phonehome(flow.data);
+        events.message(flowid, "attempted-phonehome", flow.data);
+      } else {
+        events.message(flowid, "attempted-phonehome", flow.data);
+      }
+    } else {
+      events.message(flowid, "simulated-phonehome", flow.data);
+    }
+  };
+
+  // make and setup flow
+  let flow = new Flow(local);  // create and update
+  flow.began();
+  maybePhonehome(flow);
+  */
+
+  storeFlow({data:{}});
+  events.message(local.flow_id, "began", {});
+
+  // Add parameters to url
+  let fullUrl = `${BRANCH.url}?source=hb&hbv=${VERSION}` +
+      `&c=${state.updateChannel}&v=${state.fxVersion}&l=${state.locale}` +
+      `&b=${BRANCH.name}`;
+  setTimeout(function() {
+    UITour.showHeartbeat(
+      BRANCH.prompt,
+      BRANCH.thankyou,
+      flowid,
+      fullUrl,
+      null, // learn more text, TODO: should this be "What's this?" or something?
+      null, // learn more link
+      {
+        engagementButtonLabel: BRANCH.button
+      }
+    )},
+    delay
+  );
+  return Promise.resolve(local.flow_id);
```

Comment:
This is straight out of PBM BTW.  Not that that guarantees that I did the right thing :)